### PR TITLE
chore: add read replica support to queryPaginated

### PR DESCRIPTION
**Pull request by rebelchris**
Read replica support for queryPaginated.
Pretty sure it's the most safe one to implement on as it should only be singular reads without mutations happening before.

### CODE REVIEW

**Review by idoshamun — APPROVED** [review]

File: src/common/datePageGenerator.ts [hunk 1]
```diff
@@ -93,5 +93,7 @@ export function queryPaginatedByDate<
       }
       return builder;
     },
+    undefined,
```
Comment (capJavert) [inline src/common/datePageGenerator.ts:96]: not sure about this syntax, its gonna look bad

Comment (capJavert) [reply src/common/datePageGenerator.ts:96]: i might have an idea for something else, we can try monday

Comment (rebelchris) [reply src/common/datePageGenerator.ts:96]: Decided to do as separate PR

**Review by capJavert — APPROVED** [review]
Good from my end, just check the test.

**Pull request merged**